engine/src/main/java/org/hibernate/validator/internal/util/ExecutableHelper.java

```diff
@@ -181,10 +179,26 @@ public static String getSignature(Executable executable) {
 		return getSignature( getSimpleName( executable ), executable.getParameterTypes() );
 	}
 
+	/**
+	 * That method needs to be fast.
+	 * <p>
+	 * The purpose of it is to get uniqueness.
+	 */
 	public static String getSignature(String name, Class<?>[] parameterTypes) {
-		return Stream.of( parameterTypes )
-			.map( t -> t.getName() )
-			.collect( Collectors.joining( ",", name + "(", ")" ) );
+		StringBuilder signature = new StringBuilder( name.length() + 2 + parameterTypes.length * 25 );
+		signature.append( name ).append( "(" );
```

Comment:
could use `'('`

Reply:
(I mean single quotes so to use the specialized appender method for a single char)